accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,514 @@
+
+##### What does [fulcio.sigstore.dev](https://docs.sigstore.dev/fulcio/overview/) give us?
+- Independent party to validate claims of the OIDC identity token which contains references back to the repo, workflow run and git SHA.
+- A supported CI/CD provider can't falsify these claims and no one else can claim to be one of these CI/CD providers.
+- The npm build script can't forge the source provenance by, for example, saying it's from some other repo or git SHA.
+- Vendor-neutral: can support any standards-compliant OIDC Identity Provider that can meet their requirements.
+
+##### What does [rekor.sigstore.dev](https://docs.sigstore.dev/rekor/overview/) give us?
+- Verifies that the signing happened during the time the certificate was valid and that the response from [Rekor](https://docs.sigstore.dev/rekor/overview/) is signed and can be verified offline.
+- Verifiable, append-only ledger, makes it hard to tamper with already signed packages.
+- Verification won't rely on [Rekor](https://docs.sigstore.dev/rekor/overview/) as we'll store detached signatures on the registry for performance and reliability but it adds an extra layer of protection in case of registry compromise.
+
+#### How should maintainers authenticate to sign packages?
+[Sigstore](https://www.sigstore.dev/) issues short-lived “disposable keys” notarized against log-ins with OIDC identities. So what OIDC identity provider (IdP) do maintainers use?
+
+This has been the subject of some controversy; for instance, in the [proposal to add Sigstore signing to RubyGems](https://github.com/rubygems/rfcs/pull/37), users worried about “vendorization” (making the repository reliant on third parties to function, e.g. GitHub, Google or Microsoft) and privacy (some maintainers are pseudonymous; Sigstore certificates include email addresses by default when signing using the [cosign](https://github.com/sigstore/cosign) CLI tool).
+
+Sigstore supports both email and CI/CD workload-based OIDC identities:
+
+##### Email based identity provider
+For instance, Google, GitHub, Microsoft (supported by Sigstore today) or any other IdP that Sigstore might support.
+
+**Pros**
+- Separate party for trust: it would be difficult for an attacker to compromise the repository and one of these IdPs.
+- Many (but not all!) users have these accounts already.
+- Portable: supports publishing from most environments (requiring a browser and an IdP login).
+- Minimal changes to existing publish workflow for maintainers that are publishing directly.
+- Reliability: these IdPs already work at scale and have full-time support.
+
+**Cons**
+- Privacy: the identities used to sign packages are public (though users can make a pseudonymous account).
+- Privacy: Only a handful of identity providers are supported today that might not suit all users (GitHub, Google or Microsoft).
+- Security: Some users might only have secondary, less secure, accounts on one of the supported services making it an easier target.
```

Comment:
Agreed, no way to check that a user has MFA enabled on one of those accounts for example, as this isn't included in identity tokens.

Reply:
IMO this should borderline be a requirement. I'm amazed that this isn't supported. If my publishing email is `j@gmail.com` but I publish the password publicly, this means anyone could sign in to that email and publish as me which... fundamentally gets around the entire point of this.

Reply:
A concern that has been brought up in the past is including the MFA status in identity tokens makes it easier to find targets. It would probably be simpler for all identity providers to require MFA.

Reply:
> It would probably be simpler for all identity providers to require MFA.

No opposition from me :)